src/pybind/mgr/dashboard/controllers/orchestrator.py

```diff
@@ -69,6 +76,29 @@ class Orchestrator(RESTController):
     def status(self):
         return OrchClient.instance().status()
 
+    @Endpoint(method='POST')
+    @UpdatePermission
+    @raise_if_no_orchestrator
+    @handle_orchestrator_error('osd')
+    @orchestrator_task('identify_device', ['{hostname}', '{device}'])
+    def identify_device(self, hostname, device, duration):
+        # type: (str, str, int) -> None
+        """
+        Identify a device by switching on the device light for N seconds.
+        :param hostname: The hostname of the device to process.
+        :param device: The device identifier to process, e.g. ``ABC1234DEF567-1R1234_ABC8DE0Q``.
+        :param duration: The duration in seconds how long the LED should flash.
+        """
+        orch = OrchClient.instance()
+        TaskManager.current_task().set_progress(0)
+        orch.blink_device_light(hostname, device, 'ident', True)
+        for i in range(int(duration)):
+            percentage = int(round(i / float(duration) * 100))
+            TaskManager.current_task().set_progress(percentage)
+            time.sleep(1)
```

Comment:
Progress events??? Can you please go into more detail.

Reply:
We have a dedicated "progress" module in the manager that can be used for this, can't it?

Reply:
I was not aware of that module, but it looks to me it can't be used out-of-the-box for this at them moment because it simply does not support that type of usage.

Reply:
I can start working on a follow up PR that will use the progress module infrastructure, but i don't want to change the current PR because i want to see it merged before the end of the week. Right at the moment it is not really important how it is implemented, more important is that it is in and can be tested by others on real hardware. There already popped up issues in the orchestrator that block this feature are absolutely not related by this PR.

Reply:
I've adapted the code to use the progress module. While testing i realized that the Orchestrator or any other layer also uses the progress module and reports every call of the `blink_device_light` function (twice in this case, to switch LED on and off).
```
[Complete]: blink_device_light (00h 00m 03s)
    [============================] (time remaining: N/A)
[Complete]: Identifying device /dev/vda on host host1 (00h 01m 00s)
    [============================] (time remaining: N/A)
[Complete]: blink_device_light (00h 00m 03s)
    [============================] (time remaining: N/A)
```